e2e/src/helpers/data-helper.ts

```diff
@@ -34,13 +39,48 @@ export function getControllerContent(controller: Controller): Content[] {
     .map(it => it[0]) as Content[];
 }
 
+let cachedHealthCheck: HealthCheck = null;
+
+export async function getShopHealthCheck(options?: { cache: boolean }): Promise<HealthCheck> {
+  const {cache} = R.mergeLeft(options, {cache: true})
+  let healthCheck: HealthCheck;
+  if (cache && cachedHealthCheck) {
+    healthCheck = cachedHealthCheck;
+  } else {
+    const res = await axios.get<HealthCheck>(
+      `${testConfig.prestashopUrl}/index.php?fc=module&module=ps_eventbus&controller=apiHealthCheck`
+    );
+    healthCheck = res.data;
+    cachedHealthCheck = healthCheck;
+  }
+  return healthCheck;
+}
+
+const FIXTURE_DIR = './src/fixtures';
+
 export async function loadFixture(controller: Controller): Promise<PsEventbusSyncUpload[]> {
   const contents = getControllerContent(controller);
+  const shopVersion = (await getShopHealthCheck()).prestashop_version;
   const fixture = [];
 
-  const files = contents.map(content => fs.promises.readFile(`./src/fixtures/${controller}/${content}.json`, 'utf-8'));
+  const fixtureVersions = await fs.promises.readdir(
+    `${FIXTURE_DIR}`,
+    {encoding: 'utf-8', withFileTypes: true}
+  )
+
+  // use either fixture specific to PS version or latest fixture
+  const useFixture = fixtureVersions
```

Comment:
this is the main course of this PR : we use different fixtures to be able to test our module against many versions of prestashop at once.